src/stan/analyze/mcmc/compute_potential_scale_reduction.hpp

```diff
@@ -124,6 +275,40 @@ inline double compute_potential_scale_reduction(
   return compute_potential_scale_reduction(draws, sizes);
 }
 
+/**
+ * Computes the potential scale reduction (Rhat) using rank based diagnostic for
+ * the specified parameter across all kept samples. Based on paper
+ * https://arxiv.org/abs/1903.08008
+ *
+ * When the number of total draws N is odd, the (N+1)/2th draw is ignored.
+ *
+ * See more details in Stan reference manual section "Potential
+ * Scale Reduction". http://mc-stan.org/users/documentation
+ *
+ * Current implementation assumes draws are stored in contiguous
+ * blocks of memory.  Chains are trimmed from the back to match the
+ * length of the shortest chain.
```

Comment:
You use [`split_chains`](https://github.com/stan-dev/stan/blob/51ad447d6965a59da24c6365c078450bdaebea9b/src/stan/analyze/mcmc/split_chains.hpp#L25) which also assumes each chain is the same length